[Bug] TextColorToGenerator produces error for generic views

### Description

After adding `CommunityToolkit.Maui` to a project with generic view I'm no longer able to build it with following error

### Stack Trace

/Users/pa/GenericViewBug/CommunityToolkit.Maui.SourceGenerators/CommunityToolkit.Maui.SourceGenerators.Generators.TextColorToGenerator/EnumPickerTextColorTo_e8fe61f7-6659-4d2d-8617-dff3e09df8e7.g.shared.cs(23,62): error CS0305: Using the generic type 'EnumPicker<T>' requires 1 type arguments [/Users/pa/GenericViewBug/GenericViewBug.csproj]

### Link to Reproduction Sample

[repro](https://github.com/anpin/GenericViewBug)
You can find generated files inside the  `Generated` folder

### Steps to Reproduce

1.  git clone https://github.com/anpin/GenericViewBug
2. cd GenericViewBug
3. dotnet build

### Expected Behavior
project builds
### Actual Behavior
error
### Basic Information

- Version with issue: 1.0.0-rc3
- Last known good version: - 
- SDK: 
```
dotnet --info
.NET SDK (reflecting any global.json):
 Version:   6.0.300
 Commit:    8473146e7d

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  12.3
 OS Platform: Darwin
 RID:         osx.12-x64
 Base Path:   /usr/local/share/dotnet/sdk/6.0.300/

Host (useful for support):
  Version: 6.0.5
  Commit:  70ae3df4a6

.NET SDKs installed:
  3.1.418 [/usr/local/share/dotnet/sdk]
  3.1.419 [/usr/local/share/dotnet/sdk]
  5.0.407 [/usr/local/share/dotnet/sdk]
  5.0.408 [/usr/local/share/dotnet/sdk]
  6.0.202 [/usr/local/share/dotnet/sdk]
  6.0.300 [/usr/local/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 3.1.24 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.25 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.16 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.17 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.4 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.5 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.24 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.25 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.16 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.17 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.4 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.5 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
```
### Workaround
Remove package or disable the generator with 
```
<Target Name="DisableAnalyzers"
     BeforeTargets="CoreCompile">
        <ItemGroup>
            <Analyzer Remove="@(Analyzer)" Condition="'%(Filename)' == 'CommunityToolkit.Maui.SourceGenerators'" />
        </ItemGroup>
</Target>
```

### Reproduction imagery

<!-- If it is a visual issue, please include imagery (GIF/Screenshots) showing the problem -->

--- a/src/CommunityToolkit.Maui.SourceGenerators/Extensions/NamespaceSymbolExtensions.cs
+++ b/src/CommunityToolkit.Maui.SourceGenerators/Extensions/NamespaceSymbolExtensions.cs
@@ -33,9 +33,9 @@ static class NamespaceSymbolExtensions
 
 		while (baseType is not null)
 		{
-			var result = namedSymbolList.Any(x => x.Equals(baseType, SymbolEqualityComparer.Default));
+			var doesListContainBaseType = namedSymbolList.Any(x => x.Equals(baseType, SymbolEqualityComparer.Default));
 
-			if (result)
+			if (doesListContainBaseType)
 			{
 				return true;
 			}

--- a/src/CommunityToolkit.Maui.SourceGenerators/Extensions/NamespaceSymbolExtensions.cs
+++ b/src/CommunityToolkit.Maui.SourceGenerators/Extensions/NamespaceSymbolExtensions.cs
@@ -26,4 +26,23 @@ static class NamespaceSymbolExtensions
 			}
 		}
 	}
+
+	public static bool ContainsSymbolBaseType(this IEnumerable<INamedTypeSymbol> namedSymbolList, INamedTypeSymbol symbol)
+	{
+		INamedTypeSymbol? baseType = symbol.BaseType;
+
+		while (baseType is not null)
+		{
+			var result = namedSymbolList.Any(x => x.Equals(baseType, SymbolEqualityComparer.Default));
+
+			if (result)
+			{
+				return true;
+			}
+
+			baseType = baseType.BaseType;
+		}
+
+		return false;
+	}
 }

--- a/src/CommunityToolkit.Maui.SourceGenerators/Generators/TextColorToGenerator.cs
+++ b/src/CommunityToolkit.Maui.SourceGenerators/Generators/TextColorToGenerator.cs
@@ -74,9 +74,9 @@ class TextColorToGenerator : IIncrementalGenerator
 			// If the control is inherit from a Maui control that implements ITextStyle
 			// We don't need to generate a extension method for it.
 			// We just generate a method if the Control is a new implementation of ITextStyle and IAnimatable
-			var isMauiBaseType = IsMauiBaseType(declarationSymbol, mauiTextStyleImplementors);
+			var doesContainSymbolBaseType = mauiTextStyleImplementors.ContainsSymbolBaseType(declarationSymbol);
 
-			if (!isMauiBaseType
+			if (!doesContainSymbolBaseType
 				&& declarationSymbol.AllInterfaces.Contains(textStyleSymbol, SymbolEqualityComparer.Default)
 				&& declarationSymbol.AllInterfaces.Contains(iAnimatableSymbol, SymbolEqualityComparer.Default))
 			{
@@ -86,6 +86,7 @@ class TextColorToGenerator : IIncrementalGenerator
 					context.ReportDiagnostic(diag);
 					continue;
 				}
+
 				var nameSpace = declarationSymbol.ContainingNamespace.ToDisplayString();
 
 				var accessModifier = GetClassAccessModifier(declarationSymbol);
@@ -191,21 +192,4 @@ namespace " + textStyleClass.Namespace + @";
 		Accessibility.Internal => "internal",
 		_ => string.Empty
 	};
-
-	static bool IsMauiBaseType(INamedTypeSymbol symbol, IEnumerable<INamedTypeSymbol> mauiTypes)
-	{
-		INamedTypeSymbol? baseType = symbol.BaseType;
-		while (baseType is not null)
-		{
-			var result = mauiTypes.Any(x => x.Equals(baseType, SymbolEqualityComparer.Default));
-
-			if (result)
-			{
-				return true;
-			}
-
-			baseType = baseType.BaseType;
-		}
-		return false;
-	}
 }

--- a/src/CommunityToolkit.Maui.SourceGenerators/Generators/TextColorToGenerator.cs
+++ b/src/CommunityToolkit.Maui.SourceGenerators/Generators/TextColorToGenerator.cs
@@ -71,7 +71,13 @@ class TextColorToGenerator : IIncrementalGenerator
 				continue;
 			}
 
-			if (declarationSymbol.AllInterfaces.Contains(textStyleSymbol, SymbolEqualityComparer.Default)
+			// If the control is inherit from a Maui control that implements ITextStyle
+			// We don't need to generate a extension method for it.
+			// We just generate a method if the Control is a new implementation of ITextStyle and IAnimatable
+			var isMauiBaseType = IsMauiBaseType(declarationSymbol, mauiTextStyleImplementors);
+
+			if (!isMauiBaseType
+				&& declarationSymbol.AllInterfaces.Contains(textStyleSymbol, SymbolEqualityComparer.Default)
 				&& declarationSymbol.AllInterfaces.Contains(iAnimatableSymbol, SymbolEqualityComparer.Default))
 			{
 				if (declarationSymbol.ContainingNamespace.IsGlobalNamespace)
@@ -185,4 +191,21 @@ namespace " + textStyleClass.Namespace + @";
 		Accessibility.Internal => "internal",
 		_ => string.Empty
 	};
+
+	static bool IsMauiBaseType(INamedTypeSymbol symbol, IEnumerable<INamedTypeSymbol> mauiTypes)
+	{
+		INamedTypeSymbol? baseType = symbol.BaseType;
+		while (baseType is not null)
+		{
+			var result = mauiTypes.Any(x => x.Equals(baseType, SymbolEqualityComparer.Default));
+
+			if (result)
+			{
+				return true;
+			}
+
+			baseType = baseType.BaseType;
+		}
+		return false;
+	}
 }

--- a/src/CommunityToolkit.Maui.SourceGenerators/Generators/TextColorToGenerator.cs
+++ b/src/CommunityToolkit.Maui.SourceGenerators/Generators/TextColorToGenerator.cs
@@ -175,7 +175,7 @@ namespace " + textStyleClass.Namespace + @";
 }";
 			var source = textColorToBuilder.ToString();
 			SourceStringExtensions.FormatText(ref source, options);
-			context.AddSource($"{textStyleClass.ClassName}TextColorTo_{Guid.NewGuid()}.g.shared.cs", SourceText.From(source, Encoding.UTF8));
+			context.AddSource($"{textStyleClass.ClassName}TextColorTo.g.shared.cs", SourceText.From(source, Encoding.UTF8));
 		}
 	}
 

--- a/src/CommunityToolkit.Maui.UnitTests/Extensions/TextColorTo_Tests.cs
+++ b/src/CommunityToolkit.Maui.UnitTests/Extensions/TextColorTo_Tests.cs
@@ -164,20 +164,10 @@ namespace CommunityToolkit.Maui.UnitTests.Extensions.TextStyle
 
 	class BrandNewControl : View, ITextStyle, IAnimatable
 	{
+		public double CharacterSpacing { get; } = 0;
+
 		public Color TextColor { get; set; } = Colors.Transparent;
 
 		public Font Font { get; set; }
-
-		public double CharacterSpacing => 0;
-
-		public void BatchBegin()
-		{
-
-		}
-
-		public void BatchCommit()
-		{
-
-		}
 	}
 }

--- a/src/CommunityToolkit.Maui.UnitTests/Extensions/TextColorTo_Tests.cs
+++ b/src/CommunityToolkit.Maui.UnitTests/Extensions/TextColorTo_Tests.cs
@@ -87,6 +87,42 @@ namespace CommunityToolkit.Maui.UnitTests.Extensions
 			await Assert.ThrowsAsync<ArgumentNullException>(() => label.TextColorTo(null));
 #pragma warning restore CS8625 // Cannot convert null literal to non-nullable reference type.
 		}
+
+		[Fact]
+		public async Task GenericPickerShouldUseThePickerExtension()
+		{
+			var genericPicker = new MyGenericPicker<string>();
+			genericPicker.EnableAnimations();
+
+			Color updatedTextColor = Colors.Yellow;
+			var isSuccessful = await genericPicker.TextColorTo(updatedTextColor);
+
+			Assert.True(isSuccessful);
+		}
+
+		[Fact]
+		public async Task MoreGenericPickerShouldUseThePickerExtension()
+		{
+			var genericPicker = new MoreGenericPicker<string>();
+			genericPicker.EnableAnimations();
+
+			Color updatedTextColor = Colors.Yellow;
+			var isSuccessful = await genericPicker.TextColorTo(updatedTextColor);
+
+			Assert.True(isSuccessful);
+		}
+		
+		[Fact]
+		public async Task BrandNewControlShouldHaveHisOwnExtensionMethod()
+		{
+			var brandNewControl = new BrandNewControl();
+			brandNewControl.EnableAnimations();
+
+			Color updatedTextColor = Colors.Yellow;
+			var isSuccessful = await brandNewControl.TextColorTo(updatedTextColor);
+
+			Assert.True(isSuccessful);
+		}
 	}
 }
 
@@ -114,5 +150,34 @@ namespace CommunityToolkit.Maui.UnitTests.Extensions.TextStyle
 	interface ICustomTextStyle : ITextStyle
 	{
 
+	}
+
+	class MyGenericPicker<T> : Picker
+	{
+
+	}
+
+	class MoreGenericPicker<T> : MyGenericPicker<T>
+	{
+
+	}
+
+	class BrandNewControl : View, ITextStyle, IAnimatable
+	{
+		public Color TextColor { get; set; } = Colors.Transparent;
+
+		public Font Font { get; set; }
+
+		public double CharacterSpacing => 0;
+
+		public void BatchBegin()
+		{
+
+		}
+
+		public void BatchCommit()
+		{
+
+		}
 	}
 }

